anta/tests/snmp.py

```diff
@@ -339,3 +342,117 @@ def test(self) -> None:
             self.result.is_success()
         else:
             self.result.is_failure(f"The following SNMP error counters are not found or have non-zero error counters:\n{error_counters_not_ok}")
+
+
+class VerifySNMPNotificationHost(AntaTest):
+    """Verifies the SNMP notification host (SNMP manager) configurations.
+
+    This test performs the following checks for each specified host:
+
+     1. Verifies that the SNMP host and hostname is configured on the device.
+     2. Verifies that the notification type matches the expected value.
+     3. Ensures that UDP port provided matches the expected value.
+     4. Ensures that the a valid community string is properly set for SNMP version v1/v2c and it should matches the expected value.
+     5. Ensures that the a valid user field is properly set for SNMP version v3 and it should matches the expected value.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all of the following conditions are met:
+        - The SNMP host and hostname is configured on the device.
+        - The notification type and UDP port matches the expected value.
+        - The valid community string is properly set for SNMP version v1/v2c and it should matches the expected value.
+        - The valid user field is included for SNMP v3 and it should matches the expected value.
```

Comment:
why `host AND hostname`?

```suggestion
    * Success: The test will pass if all of the following conditions are met:
        - The SNMP host and hostname is configured on the device.
        - The notification type and UDP port match the expected value.
        - Either
            - The community string is set for SNMP version v1/v2c and it matches the expected value.
            - The user field is set for SNMP v3 and it matches the expected value.
```